contrib/src/main/scala/akka/stream/contrib/SourceGen.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import akka.stream._
+import akka.stream.scaladsl._
+import akka.stream.stage._
+
+/**
+ * Source factory methods are placed here
+ */
+object SourceGen {
+
+  /**
+   * Create a `Source` that will unfold a value of type `S` by
+   * passing it through a flow. The flow should emit a
+   * pair of the next state `S` and output elements of type `E`.
+   * Source completes when the flow completes.
```

Comment:
Document in both cases that non 1-to-1 flows will simply deadlock this Source (i.e. no dropping or "multiplying" elements are supported).